### ISSUE TITLE
'ip: ioctl 0x8913 failed: No such device' after changing to virt

### ISSUE THREAD

**Issue by IanS94** [mcrute/alpine-ec2-ami#32]
Hi all,

We've started seeing the error in the title on instances stood up with the latest image. We do add in python3, pip and AWSCLI to the image make, along with gcc and some basis build libs although wouldn't imagine these being an issue for the interface name 'eth0' not being found

Is there anything you're aware of that could cause this, tried it with NIC attachment and inline with terraform to check if the interface was up at launch time (doesn't make sense but trying anything) and both have the same result. 

![image](https://user-images.githubusercontent.com/30799207/51306832-28687000-1a36-11e9-8d72-3f92bf6c1d2d.png)

Found this which kind of explains what it means, just not sure whats missing
https://www.artwork.com/support/flexlm/ioctl_failed_error.htm

If it makes any difference, this is on a C5.large instance type, so will be making use of the SRVIO and ENA

Thanks in advance

**Comment by IanS94** [comment]
Just turned off the instance, swapped it to c4.large and its up and working with udhcpc working..

Looks like it's ENA/SRVIO related

**Comment by tomalok** [comment]
Interesting.  I had the same/similar thing happen to me on a t3.nano I spun up for testing a new AMI (but then didn't get back to until a day later).  I thought it might be an EC2 hiccup or something, and either rebooted or terminated/recreated (can't recall at present) and it was fine the next time around.

I'll take a look into this when I have a chance (hopefully this weekend).

**Comment by tomalok** [comment]
I've confirmed the issue.

Alpine Linux's `linux-virt` for edge (and 3.9, due soon) is 4.19.x, and includes the `ena` module.  For 3.8, however, we used to have `aws-ena-driver` in the edge-testing repo -- but that's no longer available, and 3.8's `linux-virt` is 4.14.x -- if the `ena` driver is available there, it's not the most recent version.

To resolve this, I could either modify the AMI build to also download/compile/install an updated `ena` driver (a cleaner 3.8) or (more easily) install the edge/3.9rc kernel instead (which is something that's been done previously).
